### DISCUSSION TITLE
Issue templates: specify a project

### DISCUSSION THREAD

**Discussion by mikermcneil** [community/community#9687] [Projects and Issues] [answered]
## User story
As a GitHub user defining an issue template, I want to specify a project for that issue template, just like I can specify labels, assignees, and so forth.

## Depth
For my use case, it's not a big deal to be able to define multiple projects, or even a specific column within the project.  (The project's default column works ok.  Being able to specify a column would be nice, but it's not essential to get value from this.)

## Screenshot
![image](https://user-images.githubusercontent.com/618009/148331190-4fa6faf3-9126-4f03-ab96-f2bdfe05c183.png)

## Use case
Context: At Fleet, we are active users of GitHub Projects, and we're trying our best to make it work.  We've invested a lot of time in it, have grown to 21 people, and are now about to make a more permanent decision on whether to stick with projects.  We would like to!

Regardless, we are quite committed to using Github issue templates, and use them for everything from file bugs to employee onboarding to closing enterprise sales, and more.  We are getting a lot of value from issue templates!

## Questions?
Would you please reach out to me via email, LinkedIn message, or a public tweet to twitter.com/mikermcneil? (i might not see DMs)

Thank you!

**Comment by gabrielbdornas** [comment]
Hello @mikermcneil, yesterday I was just watching [this vídeo](https://www.youtube.com/watch?v=SI1ra-XHWHM) to figure out how the GitHub project works and start to create some tests in one of my repositories. Immediately I created issues templates and tried to use the existent syntax to link them with my brand new project, just like labels and assignees. Good to know that you have the same insight! Taking care of ours projects in just one place is a really good idea and Github is showing itself a perfect option, isn't it?

**Comment by numbsafari** [comment]
@mikermcneil would be curious to know if you all have stuck with it and if you'd be willing to share some insights into what is/is not working for you all.

It would be great to use GH Issue Templates for this, and/or have the ability to create a "default project" for a repo, so that any issues created in the repo are automatically added to that project. Sucks to have to burn Actions minutes on stuff like that.

**Reply by liushooter** [reply]
good idea ！

**Comment by omnika-foundation** [comment]
This is a missing albeit necessary feature.

**Comment by tomhuds** [comment]
Yep can we get this enabled ASAP please?

**Comment by Khamida-schemeverge** [comment]
Totally agreed!!! Very much needed feature!!!!

**Comment by sarpulas** [comment]
This would be extremely useful for moving our automations entirely to github projects

**Comment by ashrafguitoni** [comment]
Would be very useful to have, indeed.

**Comment by dmorrill** [comment]
I would use this, was looking for how to do this today

**Comment by simonebeets** [comment]
I'm hoping that this feature is implemented as I just looked around to see if there was a way to set a project to an issue template now and found this discussion.

**Comment by shamitpatel** [comment]
would be a truly useful feature.

**Comment by nicolasvienot** [comment]
I was also looking for a way to do this. I believe this would be an excellent addition 👍

**Comment by xrdavies** [comment]
Cannot wait for this feature! We are moving our community management on GitHub. Right now, assigning the issue to a specific project is blocking us.

**Comment by Khamida-schemeverge** [comment]
For someone who does the Onboarding using GitHub a lot, this feature is essential. The templates are already great but choosing the project every time is tiresome and sometimes I forget to assign it to the right project when I need to urgently create multiple issues.
For the sake of the smooth onboarding and other operations please make this happen in 2023! Thank you!

**Comment by hiroakiito2002** [comment]
Hi folks, happy new year.

Total overkill, but I've managed to achieve this request using [Actions](https://docs.github.com/en/actions), so I thought I'd share it.

I've used the following script to add any issues created in a repo to a single project. You can replicate the script and secret to replicate the feature in multiple repo case:

```
name: Add issues to project

on:
  issues:
    types: [opened]

jobs:
  add-to-project:
    name: Add issue to project
    runs-on: ubuntu-latest
    steps:
      - uses: actions/add-to-project@v0.4.0
        with:
          project-url: https://github.com/users/<user id>/projects/<project id>
          github-token: ${{ secrets.ADD_TO_PROJECT_PAT }}
```



Hope this helps!

**Reply by gabrielbdornas** [reply]
Awesome! I've just implemented this idea and it works fine! Thanks!

**Comment by Techboidesign** [comment]
There seems to be now a section called **Workflows** directly embedded to the **Project folder** 
There you can **trigger actions** like **if new project + label = move to board**

<img width="483" alt="image" src="https://user-images.githubusercontent.com/35201107/214390995-79ed185b-f10a-4440-ac5f-118008250660.png">

**Reply by gabrielbdornas** [reply]
Interesting!

Unfortunately, this option didn't appear to me yet.

![image](https://user-images.githubusercontent.com/49699290/214598518-50da4227-5583-44c2-b515-d2297a0ea7e6.png)

**Reply by Tilation** [reply]
It's missing for me too

**Reply by Khamida-schemeverge** [reply]
missing here as well

**Reply by yngvark** [reply]
I have it. It is exactly what I need. However, I have many repositories, and right now I can duplicate this workflow only 4 times.

When I have reached the maximum number of workflows, the duplicate-button is disabled with the hover-text

> maximum number of workflows reached. Delete a workflow to continue.

I really hope GitHub can increase this number!

**Comment by munir131** [comment]
it would be an excellent addition. Either we can use issue template or settings.yml

**Comment by BenjaminYde** [comment]
Would be a very nice feature!

**Comment by kehonda** [comment]
https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/adding-items-automatically

By using this workflow, I was able to achieve this without the ability to add projects from issue templete.
thank you.

**Comment by dt-undairstand** [comment]
I agree with this demand i'm looking for the same feature, it's uncomfortable to do it manually for each issue created.

**Comment by Aarbel** [comment]
Agree too to add this feature:

- project AND project's status definition directly from template

**Comment by alissontenorio** [comment]
Please include this feature

**Comment by daltonmainilnutrien** [comment]
Adding a comment in support of this feature, would be very useful.

**Comment by rkist** [comment]
Please I need this!!

**Comment by joeppeeters** [comment]
I just found out about a new toplevel property you can set in your template definition:

> *projects* (Array or String): This issue will be automatically added to these projects. Can be array of projects or comma-delimited string, e.g. "github/1,github/2"

This is from the documentation drawer which appears when you open the template for editing in the GH web editor. I did not found this in the docs elsewhere.

The project board references are formatted as `<org>/<project_id>`.


Example `.github/ISSUE_TEMPLATE/my-template.yml`:
```yaml
name: MyIssueTemplate
projects: "github/1"
body:
  - ...

```

**Reply by get6** [reply]
Yes! this information I was looking for on the web!

**Comment by gabrielbdornas** [comment]
> I just found out about a new toplevel property you can set in your template definition:
> 
> > _projects_ (Array or String): This issue will be automatically added to these projects. Can be array of projects or comma-delimited string, e.g. "github/1,github/2"
> 
> This is from the documentation drawer which appears when you open the template for editing in the GH web editor. I did not found this in the docs elsewhere.
> 
> The project board references are formatted as `<org>/<project_id>`.
> 
> Example `.github/ISSUE_TEMPLATE/my-template.yml`:
> 
> ```yaml
> name: MyIssueTemplate
> projects: "github/1"
> body:
>   - ...
> ```

This new feature is awesome! I also found [this github action to add project to issues](https://github.com/actions/add-to-project). I started to use this action before the new issue template feature and think have this two options is perfect.

**Comment by jehtee** [comment]
Is there a special trick?  i cannot get this to work, and I have tried every possible variation of project path for the "projects: " property I can think of and the new issues i create from the relevant template do not have a default project set.

Other properties like 'label' and 'assignee' do read from the front matter correctly.  But not 'project'

**Reply by vishalchangrani** [reply]
same

**Reply by sharmi** [reply]
I too tried multiple iterations and finally figured it out.

If your project url is https://github.com/orgs/OrgName/projects/1

Then you have to specify the following in the Issue Template

```
name: MyIssueTemplate
projects: "OrgName/1"
body:
```

This works for me

**Reply by jehtee** [reply]
Here is my issue created using a template with that front matter:

![image](https://github.com/community/community/assets/90803118/b600e331-3599-47f5-b5c0-cec6162e428c)

And here is the template code:

![image](https://github.com/community/community/assets/90803118/e96597c4-5472-4bcb-baa9-f869a88588b7)

And here is the address bar show the project path:

![image](https://github.com/community/community/assets/90803118/96c32efd-1edc-487c-a317-bd1d043f46de)

As you can see from the first image, this template code is not assigning new issues to the identified project.  

I'm sure that I'm doing something wrong, but I just cannot see what it is.

**Reply by traumverloren** [reply]
Sorry for the confusion around this! This feature is only supported in the `yaml` based issue forms. More info: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

**Comment by codeagencybe** [comment]
@jehtee 

Did you manage to get this working with that frontmatter for projects?
I have exactly the same problem. 
All other fields like assignee, labels, ... are mapping totally fine, but projects is always empty. 

I have checked and compared the orgname like 50x and it is 100% identical, yet it does not map. 
Feels like there is something wrong at Github side like a bug, because I don't I don't see anything else possible.

**Reply by traumverloren** [reply]
Sorry for the confusion around this! This feature is only supported in the `yaml` based issue forms. More info: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

**Comment by sharmi** [comment]
[image:
https://user-images.githubusercontent.com/90803118/259155905-6112b11c-2aa3-40e7-85ab-54ccd3623865.png]

For this project, the id to be specified is

projects: "Test-Org/29"

This way I am able to successfully add default Projects to my issues.

**Reply by codeagencybe** [reply]
I have used the correct the path like it is showing for me, but I don't have that /org part?

When I check the URL for my project, it's like this: 
https://github.com/users/codeagencybe/projects/18/

So I used "/codeagencybe/18" like:

`projects: "/codeagencybe/18"`

But it does not work while the path is absolutely correct. 


Is this a feature that only works under org repo's?

**Comment by sharmi** [comment]
You don't need the first forward slash I think. That might be the issue.

**Reply by codeagencybe** [reply]
I tried with and without /, still no difference. 
I think it's something related to organisations only or maybe a bug that's it's only checking for projects in orgs. 
Don't know, but it's annoying for sure haha

**Comment by rileybroughten** [comment, marked as answer]
You can now specify a project in your issue forms! Check out the changelog announcement [here](https://github.blog/changelog/2023-08-10-github-issues-projects-august-10th-update/#updates-to-issue-forms). If you have other feedback on issue forms and templates, feel free to drop a note in this [discussion](https://github.com/orgs/community/discussions/63402) 👍 

![image](https://github.com/community/community/assets/101840513/7e5dd036-b116-4b85-a41d-a68852391206)

**Reply by jehtee** [reply]
@rileybroughten, 

It still does not work.  This issue should not be closed.

When using a template with "projects: " defined, should the project name display in the right panel the was assignee and labels display?  or will project name be filled only after the issue has been created?

I either case, nothing has changed in my experience.    Is there some trick?

this is what my template code looks like:
![image](https://github.com/community/community/assets/90803118/557b86bb-a55c-4db6-b0c3-a888f942f835)

**Reply by tellisnz** [reply]
Hi @jehtee - this is specifically for the .yaml issue templates - it doesn't seem to work for the markdown ones with frontmatter. More docs [here](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms)

**Reply by Daffiw** [reply]
![image](https://github.com/community/community/assets/141542580/fa9620ec-a7a7-416e-873c-adbb8e598b76)

**Reply by Daffiw** [reply]
Only the finger copy's starts with Jason and Jason package in have isolation modus en .js off that seems too work a little but is see al the time orders chevron did all, this is just a little ✌🏼

**Reply by IDrumsey** [reply]
@tellisnz Do you know if they're planning on adding support for the Frontmatter md format?

**Reply by LBWright** [reply]
It's a little frustrating that this isn't supported in the MD templates, since the MD templates are the defaults when a new template is created from the github UI.

**Reply by IDrumsey** [reply]
Yeah fr, but I did find a solution. Just add the workflow for auto-adding items from the repo - https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/adding-items-automatically

**Reply by aaclayton** [reply]
How is this issue treated as complete if it does not support markdown templates?

**Reply by sfla** [reply]
> You can now specify a project in your issue forms!

This has been working for us for months, and suddenly stopped working on Wednesday last week. Any ideas why?

**Reply by rileybroughten** [reply]
Thank you for the report, this has been raised with the team and we will provide an update when it is fixed!

**Reply by hg8** [reply]
> You can now specify a project in your issue forms!

Same here, not working anymore in our org. Thanks for any update.

**Reply by rileybroughten** [reply]
We just shipped a fix for this - let us know if you are still encountering the issue 👍

**Reply by sfla** [reply]
@rileybroughten Thank you - it works perfectly again 👍

**Comment by JamesRivers** [comment]
for non org user accounts, but personal pro. What is the fix here?

**Comment by NickLarsenNZ** [comment]
I think this discussion needs to be reopened. It is only solved for YAML templates, not Markdown templates (which is what was used as an example in the original discussion description).
_Unless it is tracked in an issue somewhere._

**Comment by nbarnes22** [comment]
This is unsolved for md templates.

**Reply by Uhucream** [reply]
Me either.

**Comment by Dzivo** [comment]
Any news on this ?

**Comment by StatsRhian** [comment]
Any update on this for .md templates?
